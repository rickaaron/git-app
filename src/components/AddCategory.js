import React from 'react'
import { useState } from 'react'

export const AddCAtegory = ( { setCategorie} ) => {
    const [inputValue, setinputValue] = useState('')

    const handleSubmit = (e) =>{
        e.preventDefault();
        setCategorie( cats => [inputValue , ...cats] );
        setinputValue('');
    }
    return (
        <div>
            <h2>{inputValue}</h2>
            <form onSubmit={handleSubmit}>
                <input type="text" value={inputValue} onChange={ (e) => setinputValue(e.target.value) } />
            </form>
        </div>
    )
}
