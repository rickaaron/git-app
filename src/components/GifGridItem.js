import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( {title, url} ) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <h3>{ title }</h3>
        </div>
    )
}

GifGridItem.PropTypes={
    title : PropTypes.string.isRequired,
    url : PropTypes.string.isRequired,
}

