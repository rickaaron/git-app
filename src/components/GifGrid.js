import React from 'react'
import { useState, useEffect } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
export const GifGrid = ({title}) => {
    // const [count, setCount] = useState(0);
    // const [Images, setImages] = useState([])

    const { data , loading} = useFetchGifs(title);

    return (
        <div>
           <h3> {title} </h3>
           
           { loading ? 'Cargado...' : ''}
           <div className="container">
               {data.map( gif => <GifGridItem key={gif.id} {...gif} ></GifGridItem> )}
           </div>
        </div>
    )
}
