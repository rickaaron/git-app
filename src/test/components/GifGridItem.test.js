import { shallow } from "enzyme"
import React from 'react'
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas gif Grid Item', () => {
    let title = 'Prueba';
    let url = 'https:google.com';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> )

    test('debe mostrar el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo  ', () => {
        const parrafo = wrapper.find('h3');
        expect( parrafo.text().trim() ).toBe(title);
    })
    
    
})
