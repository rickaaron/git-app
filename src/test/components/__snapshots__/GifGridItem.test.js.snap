// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas gif Grid Item debe mostrar el componente  1`] = `
<div
  className="card"
>
  <img
    alt="Prueba"
    src="https:google.com"
  />
  <h3>
    Prueba
  </h3>
</div>
`;
